### DISCUSSION TITLE
ARC-0002: Aleo Virtual Machine (AVM)

### DISCUSSION THREAD

**Discussion by howardwu** [ProvableHQ/ARCs#5] [ARCs]
This is a discussion section for ARC-0002: Aleo Virtual Machine (AVM).

**Comment by evanmarshall** [comment]
For reference, here are the [evm opcodes](https://ethereum.org/en/developers/docs/evm/opcodes/).

The ones that immediately stick out to me as needed are:
1. LOG : a way to log data publicly. This is how you emit events in Solidity
2. ADDRESS -> `address(this)`
3. EXTCODECOPY or something similar to enable cross-contract calls
